crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -1097,22 +1113,42 @@ impl<'db> CallableBinding<'db> {
                         String::new()
                     }
                 ));
+                if let Some(function) = self.signature_type.into_function_literal() {
+                    if let Some(overloaded_function) = function.to_overloaded(context.db()) {
+                        if let Some((_, first_overload_span)) =
+                            overloaded_function.overloads[0].parameter_span(context.db(), None)
+                        {
+                            let mut sub =
+                                SubDiagnostic::new(Severity::Info, "First overload defined here");
+                            sub.annotate(Annotation::primary(first_overload_span));
+                            diag.sub(sub);
+                        }
+
+                        diag.info(format_args!(
+                            "Possible overloads for function `{}`:",
+                            function.name(context.db())
+                        ));
+                        for overload in &function.signature(context.db()).overloads.overloads {
+                            diag.info(format_args!("  {}", overload.display(context.db())));
+                        }
```

Comment:
we should maybe stop after 30 or something and print a note saying `<number> overloads omitted`? So that we don't print 300 overloads if there actually are 300 overloads?

Reply:
Yeah I was wondering about this. I think that seems sensible.

Reply:
I don't know what a reasonable cutoff is. 30? 50? 100? It probably shouldn't be more than 100 -- I don't know why you'd want more than 100 signatures printed to the terminal. The cutoff number is easy to iterate on, though, so it's probably fine to pick an arbitrary number now and add a comment saying that it's arbitrary and we can change it if we need to

Reply:
I'm not sure what the right number here is, but I feel like 300 is excessive. I think I'd maybe start at... 50? I don't know if 50 is the right number precisely, but it's hard to imagine someone carefully scrutinizing a list much bigger than that.

Reply:
SGTM

Reply:
(Also, sorry, if my comments look like they ignore more recent comments from you. GitHub used to add updated comments automatically, but now it seems to require a hard refresh of the page for me.)